### DISCUSSION TITLE
Add back button on page view and edit - filamentv3

### DISCUSSION THREAD

**Discussion by fagnerdireito** [filamentphp/filament#7705] [General]
There's a way to add a back button on the view page and on the edit page?

**Comment by gilr** [comment]
In the Resource/pages, you can add actions.
E.g. for the view, If you want to add a 'back' button that goes back to the table, before the 'Edit' button:
```
use Filament\Actions\Action;
...
class ViewCollection extends ViewRecord
{
    protected static string $resource = CollectionResource::class;

    protected function getHeaderActions(): array
    {
        return [
            Action::make('back')
                ->url(CollectionResource::getUrl()) // or you can use url(static::getResource()::getUrl())
                ->button()
                ->color('info'),
            Actions\EditAction::make(),
        ];
    }
   ...
```

What I would like to be able to do is to have this exact same button for all my views, without the need to modify all the pages.... I don't know how to do that... maybe thanks to an intermediate _MyProjectViewRecord_ class that would extend _ViewRecord_?

**Reply by dbsdenis** [reply]
> Em Resource/pages, você pode adicionar ações. Por exemplo, para a visualização, se você quiser adicionar um botão 'back' que volte para a tabela, antes do botão 'Edit':
> 
> ```
> use Filament\Actions\Action;
> ...
> class ViewCollection extends ViewRecord
> {
>     protected static string $resource = CollectionResource::class;
> 
>     protected function getHeaderActions(): array
>     {
>         return [
>             Action::make('back')
>                 ->url(CollectionResource::getUrl()) // or you can use url(static::getResource()::getUrl())
>                 ->button()
>                 ->color('info'),
>             Actions\EditAction::make(),
>         ];
>     }
>    ...
> ```
> 
> O que eu gostaria de poder fazer é ter exatamente esse mesmo botão para todas as minhas visualizações, sem a necessidade de modificar todas as páginas... Não sei como fazer isso... talvez graças a uma classe intermediária _MyProjectViewRecord_ que estenderia _ViewRecord_ ?

```
use Filament\Actions\Action;
...
class ViewCollection extends ViewRecord
{
    protected static string $resource = CollectionResource::class;

    protected function getHeaderActions(): array
    {
        return [
            Action::make('back')
                ->url(CollectionResource::getUrl()) // or you can use url(static::getResource()::getUrl())
                ->button()
                ->icon('heroicon-o-chevron-left')
                ->color('gray'),
            Actions\EditAction::make(),
        ];
    }
    ...
   ```
   
  Eu fiz assim! Obrigado...
